### DISCUSSION TITLE
newly installed spacy with gpu

### DISCUSSION THREAD

**Discussion by dx111ge** [explosion/spaCy#7898] [Help: Installation] [answered]
## How to reproduce the behaviour
(venv) C:\programming\Python\projekte\testSpacyGPU>pip install -U spacy[cuda111,transformers,lookups]
Collecting spacy[cuda111,lookups,transformers]
  Downloading spacy-3.0.6-cp38-cp38-win_amd64.whl (11.9 MB)
     |████████████████████████████████| 11.9 MB 6.4 MB/s
Collecting wasabi<1.1.0,>=0.8.1
  Using cached wasabi-0.8.2-py3-none-any.whl (23 kB)
Collecting tqdm<5.0.0,>=4.38.0
  Using cached tqdm-4.60.0-py2.py3-none-any.whl (75 kB)
Collecting jinja2
  Using cached Jinja2-2.11.3-py2.py3-none-any.whl (125 kB)
Collecting blis<0.8.0,>=0.4.0
  Using cached blis-0.7.4-cp38-cp38-win_amd64.whl (6.5 MB)
Collecting numpy>=1.15.0
  Using cached numpy-1.20.2-cp38-cp38-win_amd64.whl (13.7 MB)
Collecting preshed<3.1.0,>=3.0.2
  Using cached preshed-3.0.5-cp38-cp38-win_amd64.whl (112 kB)
Collecting pathy>=0.3.5
  Downloading pathy-0.5.2-py3-none-any.whl (42 kB)
     |████████████████████████████████| 42 kB ...
Collecting pydantic<1.8.0,>=1.7.1
  Using cached pydantic-1.7.3-cp38-cp38-win_amd64.whl (1.8 MB)
Requirement already satisfied: setuptools in c:\programming\python\projekte\testspacygpu\venv\lib\site-packages (from spacy[cuda111,lookups,transformers]) (56.0.0)
Collecting thinc<8.1.0,>=8.0.3
  Using cached thinc-8.0.3-cp38-cp38-win_amd64.whl (1.0 MB)
Collecting murmurhash<1.1.0,>=0.28.0
  Using cached murmurhash-1.0.5-cp38-cp38-win_amd64.whl (21 kB)
Collecting typer<0.4.0,>=0.3.0
  Using cached typer-0.3.2-py3-none-any.whl (21 kB)
Collecting requests<3.0.0,>=2.13.0
  Using cached requests-2.25.1-py2.py3-none-any.whl (61 kB)
Collecting packaging>=20.0
  Using cached packaging-20.9-py2.py3-none-any.whl (40 kB)
Collecting cymem<2.1.0,>=2.0.2
  Using cached cymem-2.0.5-cp38-cp38-win_amd64.whl (36 kB)
Collecting srsly<3.0.0,>=2.4.1
  Using cached srsly-2.4.1-cp38-cp38-win_amd64.whl (451 kB)
Collecting catalogue<2.1.0,>=2.0.3
  Using cached catalogue-2.0.3-py3-none-any.whl (16 kB)
Collecting spacy-legacy<3.1.0,>=3.0.4
  Downloading spacy_legacy-3.0.5-py2.py3-none-any.whl (12 kB)
Collecting cupy-cuda111<9.0.0,>=5.0.0b4
  Using cached cupy_cuda111-8.5.0-cp38-cp38-win_amd64.whl (69.4 MB)
Collecting spacy-lookups-data<1.1.0,>=1.0.0
  Using cached spacy_lookups_data-1.0.0-py2.py3-none-any.whl (93.4 MB)
Collecting spacy-transformers<1.1.0,>=1.0.1
  Using cached spacy_transformers-1.0.2-py2.py3-none-any.whl (39 kB)
Collecting fastrlock>=0.3
  Using cached fastrlock-0.6-cp38-cp38-win_amd64.whl (27 kB)
Collecting pyparsing>=2.0.2
  Using cached pyparsing-2.4.7-py2.py3-none-any.whl (67 kB)
Collecting smart-open<4.0.0,>=2.2.0
  Using cached smart_open-3.0.0-py3-none-any.whl
Collecting urllib3<1.27,>=1.21.1
  Using cached urllib3-1.26.4-py2.py3-none-any.whl (153 kB)
Collecting idna<3,>=2.5
  Using cached idna-2.10-py2.py3-none-any.whl (58 kB)
Collecting certifi>=2017.4.17
  Using cached certifi-2020.12.5-py2.py3-none-any.whl (147 kB)
Collecting chardet<5,>=3.0.2
  Using cached chardet-4.0.0-py2.py3-none-any.whl (178 kB)
Collecting transformers<4.6.0,>=3.4.0
  Using cached transformers-4.5.1-py3-none-any.whl (2.1 MB)
Collecting spacy-alignments<1.0.0,>=0.7.2
  Using cached spacy_alignments-0.8.3-cp38-cp38-win_amd64.whl (179 kB)
Collecting torch>=1.5.0
  Using cached torch-1.8.1-cp38-cp38-win_amd64.whl (190.5 MB)
Collecting typing-extensions
  Using cached typing_extensions-3.7.4.3-py3-none-any.whl (22 kB)
Collecting tokenizers<0.11,>=0.10.1
  Using cached tokenizers-0.10.2-cp38-cp38-win_amd64.whl (2.0 MB)
Collecting filelock
  Using cached filelock-3.0.12-py3-none-any.whl (7.6 kB)
Collecting regex!=2019.12.17
  Using cached regex-2021.4.4-cp38-cp38-win_amd64.whl (270 kB)
Collecting sacremoses
  Using cached sacremoses-0.0.45-py3-none-any.whl (895 kB)
Collecting click<7.2.0,>=7.1.1
  Using cached click-7.1.2-py2.py3-none-any.whl (82 kB)
Collecting MarkupSafe>=0.23
  Using cached MarkupSafe-1.1.1-cp38-cp38-win_amd64.whl (16 kB)
Collecting six
  Using cached six-1.15.0-py2.py3-none-any.whl (10 kB)
Collecting joblib
  Using cached joblib-1.0.1-py3-none-any.whl (303 kB)
Installing collected packages: urllib3, idna, chardet, certifi, requests, numpy, murmurhash, cymem, click, catalogue, wasabi, typer, tqdm, srsly, smart-open, six, regex, pyparsing, pydantic, preshed, MarkupS
afe, joblib, blis, typing-extensions, tokenizers, thinc, spacy-legacy, sacremoses, pathy, packaging, jinja2, filelock, transformers, torch, spacy-alignments, spacy, fastrlock, spacy-transformers, spacy-looku
ps-data, cupy-cuda111
Successfully installed MarkupSafe-1.1.1 blis-0.7.4 catalogue-2.0.3 certifi-2020.12.5 chardet-4.0.0 click-7.1.2 cupy-cuda111-8.5.0 cymem-2.0.5 fastrlock-0.6 filelock-3.0.12 idna-2.10 jinja2-2.11.3 joblib-1.0.
1 murmurhash-1.0.5 numpy-1.20.2 packaging-20.9 pathy-0.5.2 preshed-3.0.5 pydantic-1.7.3 pyparsing-2.4.7 regex-2021.4.4 requests-2.25.1 sacremoses-0.0.45 six-1.15.0 smart-open-3.0.0 spacy-3.0.6 spacy-alignmen
ts-0.8.3 spacy-legacy-3.0.5 spacy-lookups-data-1.0.0 spacy-transformers-1.0.2 srsly-2.4.1 thinc-8.0.3 tokenizers-0.10.2 torch-1.8.1 tqdm-4.60.0 transformers-4.5.1 typer-0.3.2 typing-extensions-3.7.4.3 urllib
3-1.26.4 wasabi-0.8.2

(venv) C:\programming\Python\projekte\testSpacyGPU>python -m spacy train config.cfg --gpu-id 0
ℹ Using GPU: 0
Traceback (most recent call last):
  File "C:\programming\Python\Interpreter\Python387\lib\runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\programming\Python\Interpreter\Python387\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\programming\Python\projekte\testSpacyGPU\venv\lib\site-packages\spacy\__main__.py", line 4, in <module>
    setup_cli()
  File "C:\programming\Python\projekte\testSpacyGPU\venv\lib\site-packages\spacy\cli\_util.py", line 69, in setup_cli
    command(prog_name=COMMAND)
  File "C:\programming\Python\projekte\testSpacyGPU\venv\lib\site-packages\click\core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "C:\programming\Python\projekte\testSpacyGPU\venv\lib\site-packages\click\core.py", line 782, in main
    rv = self.invoke(ctx)
  File "C:\programming\Python\projekte\testSpacyGPU\venv\lib\site-packages\click\core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "C:\programming\Python\projekte\testSpacyGPU\venv\lib\site-packages\click\core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "C:\programming\Python\projekte\testSpacyGPU\venv\lib\site-packages\click\core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "C:\programming\Python\projekte\testSpacyGPU\venv\lib\site-packages\typer\main.py", line 497, in wrapper
    return callback(**use_params)  # type: ignore
  File "C:\programming\Python\projekte\testSpacyGPU\venv\lib\site-packages\spacy\cli\train.py", line 51, in train_cli
    setup_gpu(use_gpu)
  File "C:\programming\Python\projekte\testSpacyGPU\venv\lib\site-packages\spacy\cli\_util.py", line 511, in setup_gpu
    require_gpu(use_gpu)
  File "C:\programming\Python\projekte\testSpacyGPU\venv\lib\site-packages\thinc\util.py", line 187, in require_gpu
    raise ValueError("GPU is not accessible. Was the library installed correctly?")
ValueError: GPU is not accessible. Was the library installed correctly?

## Your Environment
* Operating System: Windows 10
* Python Version Used: 3.8.7
* spaCy Version Used: 3
* Environment Information:

**Comment by polm** [comment, marked as answer]
Sounds like something is wrong with your GPU setup. 

1. What's your GPU?
2. What version of CUDA do you have installed?
3. Are you able to use your GPU in any other Python libraries?

It looks like you installed the cupy lib for CUDA 11.1, if that's not the right version that could be an issue.

**Reply by dx111ge** [reply]
Thanks for your help. It works in pytorch and my installed version is 11.1 (nvidia geforce RTX 2070 SUPER)

**Reply by adrianeboyd** [reply]
Hi, can you try a simple test of `cupy`?

```python
import cupy
a = cupy.zeros((1, 1))
````

**Reply by dx111ge** [reply]
okay, found it .... seems to be a problem with torch. i just reinstalled torch with the right version and it was working.... Thanks a lot! A mismatch between torch cuda and whatever else cuda

**Reply by reactivetype** [reply]
@dx111ge  I am facing same issue. could you please share the torch version and of other packages that worked for you? I appreciate if you can show your final requirements.txt. thanks
